src/metatensor/models/utils/loss.py

```diff
@@ -0,0 +1,118 @@
+from typing import Dict, Optional
+
+import torch
+from metatensor.torch import TensorMap
+
+
+# This file defines losses for metatensor models.
+
+
+class TensorMapLoss:
+    """
+    A loss function that operates on two `metatensor.torch.TensorMap`s.
+
+    The loss is computed as the sum of the loss on the block values and
+    the loss on the gradients, with weights specified at initialization.
+
+    At the moment, this loss function assumes that all the gradients
+    declared at initialization are present in both TensorMaps.
```

Comment:
Parameters (`:param reduction: ...`) are missing in the docstring.

Reply:
I would keep it this way, it's just a workaround for the fact that MT-learn doesn't have losses yet